test/e2e/csc_tests.go

```diff
@@ -131,3 +142,89 @@ func cscWithNonExistingTargetNamespace(t *testing.T, f *test.Framework, ctx *tes
 
 	return nil
 }
+
+func cscWithNonExistingPackage(t *testing.T, f *test.Framework, ctx *test.TestCtx) error {
+	// Get test namespace
+	namespace, err := ctx.GetNamespace()
+	if err != nil {
+		return fmt.Errorf("Could not get namespace: %v", err)
+	}
+
+	// nonExistingPackageCSCName is the name of the catalogsourceconfig that points
+	// to a non-existing Package
+	nonExistingPackageCSCName string = "non-existing-package-operators"
+
+	// packageName is the non-existing package
+	packageName string = "non-existing-package"
+
+	// Create a new catalogsourceconfig with a non-existing Package
+	nonExistingPackageCSC := &operator.CatalogSourceConfig{
+		TypeMeta: metav1.TypeMeta{
+			Kind: operator.CatalogSourceConfigKind,
+		}, ObjectMeta: metav1.ObjectMeta{
+			Name:      nonExistingPackageCSCName,
+			Namespace: namespace,
+		},
+		Spec: operator.CatalogSourceConfigSpec{
+			TargetNamespace: namespace,
+			Packages:        packageName,
+		}}
+	err = createRuntimeObject(f, ctx, nonExistingPackageCSC)
+	if err != nil {
+		return err
+	}
+
+	// Check that we created the catalogsourceconfig with a non-existing package
+	resultCatalogSourceConfig := &operator.CatalogSourceConfig{}
+	err = WaitForResult(t, f, resultCatalogSourceConfig, namespace, nonExistingPackageCSCName)
+	if err != nil {
+		return err
+	}
+
+	// Check if the catalogsourceconfig phase and message are the expected values
+	expectedPhase := "Configuring"
+	expectedMessage := fmt.Sprintf("Failed to find package: %v", packageName)
+	// Check that the catalogsourceconfig with an non-existing package eventually reaches the
+	// configuring phase with the expected message
+	err = wait.Poll(retryInterval, timeout, func() (bool, error) {
+		err = WaitForResult(t, f, resultCatalogSourceConfig, namespace, nonExistingPackageCSCName)
+		if err != nil {
+			return false, err
+		}
+		if resultCatalogSourceConfig.Status.CurrentPhase.Name == expectedPhase &&
+			resultCatalogSourceConfig.Status.CurrentPhase.Message == expectedMessage {
+			return true, nil
+		}
+		return false, nil
+	})
+	if err != nil {
+		return fmt.Errorf("CatalogSourceConfig never reached expected phase/message, expected %v/%v", expectedPhase, expectedMessage)
+	}
+
+	catalogSourceName := nonExistingPackageCSCName
+	serviceName := nonExistingPackageCSCName
+	deploymentName := nonExistingPackageCSCName
+
+	// Check catalog source was not created.
+	resultCatalogSource := &olm.CatalogSource{}
+	err = WaitForResult(t, f, resultCatalogSource, namespace, catalogSourceName)
```

Comment:
In this example, you seem to be using the WaitForResult to confirm that the resource doesn't exist by waiting for a timeout, which [takes 5 minutes](https://github.com/operator-framework/operator-marketplace/blob/master/test/e2e/helpers.go#L28). You seem to do this 3 times - wouldn't this add 15 minutes of checks?

Reply:
Good point - I guess since these resources are created independently I can just concurrently run these

Reply:
Is that the right approach? You would still be adding a 5 minute test to marketplace e2e tests. You could create a [new function that waits for a 404 on a resource](https://github.com/awgreene/operator-marketplace/commit/a16dc5591d766daccc83acb27cb6b5af24ad2709#diff-c92b68c95022bb062e31eb7aef425457R79)

Reply:
We could also make `waitForResult` configurable in addition.

Reply:
In fact since these resources should never exist you could just confirm that they don't exist.

Reply:
@galletti94 such re-factoring should ideally be separate PRs or at a minimum a separate commit in this PR.